src/main/java/org/terasology/gooeyDefence/PathfindingSystem.java

```diff
@@ -125,7 +130,18 @@ private void calculatePath(int id, Runnable callback) {
     }
 
     private void calculatePath(Vector3i startPoint, BiConsumer<List<Vector3i>, Vector3i> callback) {
-        pathfinderSystem.requestPath(DefenceField.fieldCentre(), startPoint, callback::accept);
+        JPSConfig config = new JPSConfig();
+        /* Backwards to ensure that zero is at the field centre */
+        config.start = startPoint;
+        config.stop = DefenceField.fieldCentre();
+        config.maxDepth = DefenceField.outerRingSize() * 2;
+        //TODO: Replace width and height with values from enemy.
+        config.plugin = new EnemyWalkingPlugin(worldProvider, 0.5f, 0.5f);
+        config.maxTime = 10f;
+        pathfinderSystem.requestPath(config, (path, end) -> {
+            Collections.reverse(path);
+            callback.accept(path, DefenceField.fieldCentre());
+        });
```

Comment:
-you can separate the config creation from the path request and callback in two methods.
-extract magic numbers to constants with speaking names, e.g. what is 10f or 0.5f
-separate defenceField access from calculation, pass these values as parameters.